include/seqan3/alignment/matrix/detail/alignment_score_matrix_proxy_type.hpp

```diff
@@ -0,0 +1,51 @@
+// -----------------------------------------------------------------------------------------------------
+// Copyright (c) 2006-2019, Knut Reinert & Freie Universität Berlin
+// Copyright (c) 2016-2019, Knut Reinert & MPI für molekulare Genetik
+// This file may be used, modified and/or redistributed under the terms of the 3-clause BSD-License
+// shipped with this file and also available at: https://github.com/seqan/seqan3/blob/master/LICENSE.md
+// -----------------------------------------------------------------------------------------------------
+
+/*!\file
+ * \brief Provides seqan3::detail::alignment_score_matrix_proxy_type.
+ * \author Rene Rahn <rene.rahn AT fu-berlin.de>
+ */
+
+#pragma once
+
+#include <seqan3/core/concept/core_language.hpp>
+#include <seqan3/core/simd/concept.hpp>
+
+namespace seqan3::detail
+{
+
+/*!\brief A proxy type for a unified access to the score matrix during alignment computation.
+ * \tparam value_type The type wrapped by this proxy.
+ * \ingroup alignment_matrix
+ *
+ * \details
+ *
+ * Provides named accessors to the respective values of the score matrix during the alignment computation.
+ * The `value_type` must either model the seqan3::Arithmetic or seqan3::detail::Simd vector concept.
+ * In some cases the last value to the left is stored in a different location then the next left value that
+ * will be used in the subsequent column (e.g. in the banded alignment matrix). Accordingly the left value is
+ * split in a reference for reading, referred to as `r_left` (for reading), and a reference to the next left value
+ * on the same row, referred to as `w_left` (for writing).
+ */
+template <typename value_type>
+struct alignment_score_matrix_proxy_type
```

Comment:
> We did not do that anywhere else

Muhhahahahaha, you need to discuss this with @h-2. He used it in several places before and I was like why? Now it falls back on me 🤣. Anyway, I gladly change this back.

Reply:
I don't remember a discussion about this? Is there a reasoning behind the type?

Reply:
Well, I can't remember completely but something about this being a type. So I usually only apply it if you would rather call your variable like that and it would otherwise clash with the type definition then. Another example is strong_type. But I can't remember having a rule or something.

Reply:
Hm but every struct is basically a type?... and it is not called `vector_type`. 
I guess `iterator_type` is to not confuse it with the `iterator` type member but who knows. Thanks for changing it I think this holds data just like the vector and should not be called type. Especially if the concept names soon end with `_type` :rofl: